helper/schema/provider.go

```diff
@@ -305,6 +329,10 @@ func (p *Provider) Resources() []terraform.ResourceType {
 		result = append(result, terraform.ResourceType{
 			Name:       k,
 			Importable: resource.Importer != nil,
+
+			// Indicates that a provider is compiled against a new enough
+			// version of core to support the GetSchema method.
+			SchemaAvailable: true,
```

Comment:
Shouldn't the protocol version indicate if we support the correct methods?

Reply:
This field is being used to avoid incrementing the protocol version so that we don't create a provider plugin compatibility wall yet. There will be some bigger changes to the provider protocol coming later, so the desire is to get it all out in a single increment so we can avoid putting users through that upgrade pain multiple times.

(If we increment the plugin protocol version then new provider releases become incompatible with older core versions, which returns us to the sad old world where you need to upgrade core in order to get the latest provider features. Tolerable to do once or twice, but not something I want to make a habit of...)

Reply:
gotcha, just making sure I understood why the workaround was here.